src/FsAutoComplete/LspServers/AdaptiveFSharpLspServer.fs

```diff
@@ -2602,26 +2607,22 @@ type AdaptiveFSharpLspServer
                     let items =
                       decls
                       |> Array.mapi (fun id d ->
-                        let code =
-                          if
-                            System.Text.RegularExpressions.Regex.IsMatch(d.NameInList, """^[a-zA-Z][a-zA-Z0-9']+$""")
-                          then
-                            d.NameInList
-                          elif d.NamespaceToOpen.IsSome then
-                            d.NameInList
-                          else
-                            FSharpKeywords.NormalizeIdentifierBackticks d.NameInList
-
-                        let label =
+                        let code = getCodeToInsert d
+
+                        let label, filterText =
                           match d.NamespaceToOpen with
-                          | Some no -> sprintf "%s (open %s)" d.NameInList no
-                          | None -> d.NameInList
+                          | Some no when config.FullNameExternalAutocomplete ->
+                            // allow filter "Ceiling (System.Math)" by "System.Math.Ceiling" or "CeilingSystem.Math"
+                            sprintf "%s (%s)" d.NameInList no, d.NameInList + code
+                          | Some no -> sprintf "%s (open %s)" d.NameInList no, d.NameInList
+                          | None -> d.NameInList, d.NameInList
 
                         { CompletionItem.Create(d.NameInList) with
+                            Data = Some(JValue(d.FullName))
                             Kind = (AVal.force glyphToCompletionKind) d.Glyph
-                            InsertText = Some code
+                            InsertText = Some(getCodeToInsert d)
```

Comment:
nit: `getCodeToInsert d` is calculate above in `code` - can we reuse it here to prevent recalculating?

Reply:
fixed